python/cudf/cudf/core/column/numerical.py

```diff
@@ -475,6 +486,42 @@ def _safe_cast_to_int(col, dtype):
         )
 
 
+def _normalize_find_and_replace_input(input_column_dtype, col_to_normalize):
```

Comment:
Not sure of a cleaner approach to handle this

Reply:
@ayushdg spent some time going through this and the approach looks good to me. The only downside I potentially see is if we're given a small list-like of replacement values we end up copying them to the GPU to calculate the min / max for dtype calculation and it would likely be faster to do it on the CPU if the input size is small. That's a super premature optimization so full steam ahead.